lib/process-services/src/lib/task-list/components/task-form/task-form.component.spec.ts

```diff
@@ -768,4 +769,198 @@ describe('TaskFormComponent', () => {
+            expect(activitFormSelector).toBeDefined();
+            expect(inputFieldOne['disabled']).toEqual(true, 'Task Form field is enabled');
+            expect(inputFieldTwo['disabled']).toEqual(true, 'Task Form field is enabled');
+            expect(inputFieldThree['disabled']).toEqual(true, 'Task Form field is enabled');
+        });
+
+        it('Should task form fields be disabled when the task has not been claimed', async () => {
+            fixture.detectChanges();
+            getTaskDetailsSpy.and.returnValue(of(claimableTaskDetailsMock));
+            component.taskId = '456';
+            component.ngOnInit();
+
+            fixture.detectChanges();
+            await fixture.whenStable();
+            fixture.detectChanges();
+
+            const activitFormSelector = element.querySelector('adf-form');
+            const inputFieldOne = fixture.debugElement.nativeElement.querySelector('#text1');
+            const inputFieldTwo = fixture.debugElement.nativeElement.querySelector('#text2');
+            const inputFieldThree = fixture.debugElement.nativeElement.querySelector('#text3');
+            expect(activitFormSelector).toBeDefined();
+            expect(inputFieldOne['disabled']).toEqual(true, 'Task Form field is enabled');
+            expect(inputFieldTwo['disabled']).toEqual(true, 'Task Form field is enabled');
+            expect(inputFieldThree['disabled']).toEqual(true, 'Task Form field is enabled');
+        });
+
+        it('Should task form fields be disabled when the task is completed', async () => {
+            fixture.detectChanges();
+            getTaskDetailsSpy.and.returnValue(of(completedTaskWithFormMock));
+            component.taskId = '456';
+            component.ngOnInit();
+            fixture.detectChanges();
+            await fixture.whenStable();
+            fixture.detectChanges();
+
+            const activitFormSelector = element.querySelector('adf-form');
+            const inputFieldOne = fixture.debugElement.nativeElement.querySelector('#text1');
+            const inputFieldTwo = fixture.debugElement.nativeElement.querySelector('#text2');
+            const inputFieldThree = fixture.debugElement.nativeElement.querySelector('#text3');
+
+            expect(activitFormSelector).toBeDefined();
+            expect(inputFieldOne['disabled']).toEqual(true);
+            expect(inputFieldTwo['disabled']).toEqual(true);
+            expect(inputFieldThree['disabled']).toEqual(true);
+        });
+
+        it('Should task form fields be enabled when the task is claimed', async() => {
+            fixture.detectChanges();
+            getBpmLoggedUserSpy.and.returnValue(of(claimedTaskDetailsMock.assignee));
+            getTaskDetailsSpy.and.returnValue(of(claimedTaskDetailsMock));
+            component.taskId = '456';
+            component.ngOnInit();
+            fixture.detectChanges();
+            await fixture.whenStable();
+            fixture.detectChanges();
```

Comment:
`         component.ngOnInit();
            fixture.detectChanges();
            await fixture.whenStable();
            fixture.detectChanges();`

Is all that needed? Same for all the other occurences

Reply:
Yes It was needed. adf-form was not detecting the changes without fixture.detectChanges(); after the whenstable called.